### PR TITLE
Add some metrics

### PR DESCRIPTION
**Pull request by huachaohuang**

### CODE REVIEW

File: src/util/rocksdb/engine_metrics.rs [hunk 1]
```diff
@@ -864,6 +1006,15 @@ pub fn flush_engine_properties(engine: &DB, name: &str) {
             }
         }
     }
+
+    // For snapshot
+    if let Some(n) = engine.get_property_int(ROCKSDB_NUM_SNAPSHOTS) {
+        STORE_ENGINE_NUM_SNAPSHOTS_GAUGE.set(n as f64);
+    }
+    if let Some(t) = engine.get_property_int(ROCKSDB_OLDEST_SNAPSHOT_TIME) {
+        let d = time::get_time().sec.checked_sub(t as i64).unwrap_or(0);
```
Comment (siddontang) [inline src/util/rocksdb/engine_metrics.rs]: the time is second?

File: src/util/rocksdb/engine_metrics.rs [hunk 2]
```diff
@@ -23,16 +24,28 @@ pub const ROCKSDB_ESTIMATE_NUM_KEYS: &'static str = "rocksdb.estimate-num-keys";
...
                                                             estimate-pending-compaction-bytes";
 pub const ROCKSDB_COMPRESSION_RATIO_AT_LEVEL: &'static str = "rocksdb.compression-ratio-at-level";
+pub const ROCKSDB_NUM_SNAPSHOTS: &'static str = "rocksdb.num-snapshots";
+pub const ROCKSDB_OLDEST_SNAPSHOT_TIME: &'static str = "rocksdb.oldest-snapshot-time";
 
 pub const ENGINE_TICKER_TYPES: &'static [TickerType] = &[
     TickerType::BlockCacheMiss,
     TickerType::BlockCacheHit,
+    TickerType::BlockCacheAdd,
+    TickerType::BlockCacheAddFailures,
```
Comment (zhangjinpeng1987) [inline src/util/rocksdb/engine_metrics.rs]: Can we get the failure reason?

Comment (siddontang) [reply src/util/rocksdb/engine_metrics.rs]: seem can't

File: src/util/rocksdb/engine_metrics.rs (hunk 1 above)
Comment (huachaohuang) [reply src/util/rocksdb/engine_metrics.rs]: Yes

**Pull request merged**